src/usuarios/usuarios-controlador.js

```diff
@@ -15,59 +15,67 @@ function criaTokenJWT(usuario) {
 }
 
 function encripta(payload, chave) {
-  chave = crypto.scryptSync(chave, 'salt', 32);
-  console.log(chave);
   // iv: initialization vector
   const vetorInicializacao = crypto.randomBytes(16);
   const encriptador = crypto.createCipheriv(
     'aes-256-gcm',
     chave,
     vetorInicializacao
   );
-  const payloadEncriptado =
-    encriptador.update(payload, 'utf8') + encriptador.final();
+
+  const payloadEncriptado = Buffer.concat([
+    encriptador.update(payload, 'utf8'),
+    encriptador.final()
+  ]);
+
   const tag = encriptador.getAuthTag();
-  const refreshToken = JSON.stringify({
-    payloadEncriptado,
-    tag,
-    vetorInicializacao
-  });
-  return Buffer.from(refreshToken).toString('base64');
+
+  // dá pra fazer sem .map() nem Buffer.concat() mas não fica tão simétrico com o decripta
+  // e essa forma facilita pra adicionar elementos no refresh token
+  const refreshToken = [payloadEncriptado, tag, vetorInicializacao]
+    .map(item => item.toString('hex'))
+    .join('.');
```

Comment:
Que outras informações seriam interessantes adicionar no refresh token? Acredito que temos que ter o mínimo de infos possíveis para não vazar nada sensível

Reply:
Na implementação que eu estava vendo dos [Super Tokens](https://supertokens.io/blog/the-best-way-to-securely-manage-user-sessions) eles colocavam várias coisas no refresh token. A maioria era informação extra pra ajudar na detecção de tokens roubados. Então, não é nada que vai ser usado no curso mesmo